books/spiders/book.py

```diff
@@ -0,0 +1,27 @@
+import scrapy
+from scrapy.http import Response
+
+
+class BookSpider(scrapy.Spider):
+    name = "book"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs):
+        for book in response.css(".image_container a::attr(href)"):
+            yield response.follow(book, callback=self._parse_single_book)
+
+        next_page = response.css(".pager .next a::attr(href)").get()
+        if next_page:
+            yield response.follow(next_page, callback=self.parse)
+
+    def _parse_single_book(self, response: Response):
+        yield {
+            "title": response.css("h1::text").get(),
+            "price": float(response.css(".price_color::text").get().replace("£", "")),
+            "amount_in_stock": int(response.css(".instock::text").re_first(r"\d+")),
```

Comment:
Ensure that the `response.css(".instock::text")` selector correctly targets the text containing the stock amount. If the text structure changes, this could lead to errors. Consider adding error handling or default values.